geps/gep-2659/index.md

```diff
@@ -0,0 +1,167 @@
+to the Gateway API spec, and MAY result in further GEPs to further clarify.
+For GEPs that _do_ make changes to the API, but also require further GEPs to
+clarify, they SHOULD use the new "Updated By" relationship instead.
+
+Memorandum GEPs should be used sparingly, and should form the umbrella for a
+significant amount of work, particularly work that may have parts that can
+move through the GEP phases at different speeds.
+
+The status is reached when a Memorandum GEP is merged, although as we will document
+in the "Addition of GEP relationships" section, it can still be Updated
+or Obsoleted.
+
+Existing GEPs that meet this criteria will be gradually moved to be proper
+Memorandum GEPs after this GEP is merged.
+
+### Addition of YAML metadata file
+
+The core Gateway API mainatainers were hoping not to need metadata YAMLs for a
+while, but the addition of relationships has turbocharged the need for machine
+parseable GEP metadata.
+
+This should also help with building display for GEPs; theoretically we can build
+tooling that will let people slice the list of GEPs by whatever dimensions
+we (or they) wish.
+
+Similarly to the `ConformanceReport` object, I'm proposing to make CRD definitions
+for this `GEPDetails` object, which will not be included in the usual CRD
+definitions for Gateway API, nor will it be part of the regular API spec.
+
+The use of a CRD is just to make the schema more similar to the other schemas
+in this repository.
+
+With that said, here's a rough sample of what a GEPMetadata object will look like:
+
+```yaml
+apiVersion: internal.gateway.networking.k8s.io/v1alpha1
+kind: GEPDetails
+number: 2659
+name: Document and improve the GEP process
+status: Memorandum
+authors:
+  - youngnick
+relationships:
+  # obsoletes indicates that a GEP makes the linked GEP obsolete, and completely
+  # replaces that GEP. The obsoleted GEP MUST have its obsoletedBy field
+  # set back to this GEP, and MUST be moved to Declined.
+  obsoletes: {}
+  obsoletedBy: {}
+  # extends indicates that a GEP extends the linkned GEP, adding more detail
+  # or additional implementation. The extended GEP MUST have its extendedBy
+  # field set back to this GEP.
+  extends: {}
+  extendedBy: {}
+  # seeAlso indicates other GEPs that are relevant in some way without being
+  # covered by an existing relationship.
+  seeAlso: {}
+# latestRelease indicates the latest release in which the GEP was changed.
+# update this when you update a GEP.
+latestRelease: v1.1
+# references is a list of hyperlinks to relevant external references.
+# It's intended to be used for storing Github discussions, Google docs, etc.
+references: {}
+# featureNames is a list of the feature names introduced by the GEP, if there
+# are any. This will allow us to track which feature was introduced by which GEP.
+featureNames: {}
+# changelog is a list of hyperlinks to PRs that make changes to the GEP, in
+# ascending date order.
+changelog: {}
```

Comment:
Should you add a self-referential link here and in the corresponding metadata.yaml to provide an example?

Reply:
Oh, good point, yes I should, thanks.